core/server/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -510,6 +512,12 @@ public void addInodeFromJournal(JournalEntry entry) {
 
       if (directory.getName().equals(ROOT_INODE_NAME)) {
         // This is the root inode. Clear all the state, and set the root.
+        if (SecurityUtils.isSecurityEnabled(MasterContext.getConf())
+            && mRoot != null && !mRoot.getUserName().equals(directory.getUserName())) {
+          // user is not the owner of journal root entry
+          throw new IOException(
```

Comment:
Good question. `addInodeFromJournal` is called by `processJournalEntry` which throws IOException.

How about having `AccessControlException` here and then catch it and convert to `RuntimeException` in `processJournalEntry` ? similar as the other Exception handlings in `processJournalEntry`

Throwing a IOException due to permission denied seems weird to me. What do you think?